bin/ansible2gdeploy.py

```diff
@@ -28,6 +32,50 @@
 GLUSTER_SERVER_GROUP = "gluster-servers"
 GLUSTER_CLIENT_GROUP = "usm_client"
 
+
+def host_fqdn(fqdn):
+    """
+    Return host's fqdn.
+    """
+    print("FQDN: %s" % fqdn, file=sys.stderr)
+    return fqdn
+
+def host_short(fqdn):
+    """
+    Return host's short hostname.
+    """
+    short = fqdn.split(".")[0]
+    print("SHORT: %s" % short, file=sys.stderr)
+    return short
+
+def host_ip(fqdn):
+    """
+    Return host's IP.
+    """
+    ip = socket.gethostbyname(fqdn)
+    print("IP: %s" % ip, file=sys.stderr)
+    return ip
+
+def host_mixed(fqdn):
+    """
+    Return host's fqdn, hostname or IP - sequentially selected.
+    """
+    choices = (host_fqdn, host_short, host_ip)
+    try:
+        current_i = host_mixed.i
+    except AttributeError:
+        host_mixed.i = 0
+        current_i = 0
+    host_mixed.i = (host_mixed.i + 1) % len(choices)
+    return choices[current_i](fqdn)
```

Comment:
Eh, I don't like this way of doing things, but let's keep it. Could you add here a comment stating something like:

```
# HACK: instead of using a global variable or iterator, we just push an attribute to the function object to preserve global state
```